app/services/towing_task_service.rb

```diff
@@ -0,0 +1,52 @@
+class TowingTaskService < TaskService
+  INACTIVE_TASK_REMINDER_TIME = ClowderConfig.instance["INACTIVE_TASK_REMINDER_TIME"].to_i * 60
+
+  def process
+    inactive_tasks.each do |task|
+      towing_task = TowingTask.create!(task_options(task))
+      towing_task.dispatch
+
+      Rails.logger.info("Towing the inactive task: #{task}")
+      task.update!(:message => "Inactive, need towing task to reactivate")
+    end
+
+    self
+  rescue => e
+    Rails.logger.error("Towing task failed: #{e.message}")
+  end
+
+  private
+
+  def inactive_tasks
+    tasks = Task.arel_table
+    LaunchJobTask.where(tasks[:created_at].lt(Time.current - INACTIVE_TASK_REMINDER_TIME).and(tasks[:state].eq("running")).and(tasks[:source_id].eq(@source.id)))
+  end
+
+  def task_options(task)
+    {}.tap do |opts|
+      opts[:child_task_id] = task.id
+      opts[:forwardable_headers] = task.forwardable_headers
+      opts[:source_id] = task.source.id
+      opts[:tenant] = task.tenant
+      opts[:state]  = 'pending'
+      opts[:status] = 'ok'
+      opts[:input] = input(task)
+    end
+  end
+
+  def input(task)
+    CatalogInventory::Payload.new('json', upload_url, jobs(task), @source.previous_sha, @source.previous_size).as_json
+  end
+
+  def jobs(task)
+    jobs = []
+
+    launch = CatalogInventory::Job.new
+    launch.href_slug = task.output["url"]
```

Comment:
@hsong-rh Its possible that the task.output doesn't have a URL. If we send the task and the worker dies before it can update the output, he wouldn't be able to set output['url']. So we need to account for that. When we create the TowingTask we should ensure that the output['url'] has a value. if it's empty I think we can safely mark the LaunchJobTask as failed. Since we would be doing this once per hour after the availability check has run.

Reply:
@hsong-rh This is akin to when we send the Tow Truck we need to know the license number.